pagopa/io-app
# feat: [IOPID-316] FL Refinement

## Short description
This PR add some refinement to the FL feature by handling `401` (refresh) and `403` (session expiration) and transient errors (courtesy screen)

## List of changes proposed in this pull request
- `backendStatus.json` synch in the `startup` saga. 
- Add an error screen if the APP receives `maxRetries` transient errors from the backend server.
- Add a loading spinner during the token refresh.
- If FL is enabled, `CheckSession` after every `IdentificationSuccess` is removed to avoid a refresh loop.

| Loading Spinner |  Error Screen <br /> on transient error during refresh |
| - | - |
| <img width="200" src="https://github.com/pagopa/io-app/assets/16268789/6f243375-df53-4943-9ec9-96ed6248a9ab" /> | <img width="200" src="https://github.com/pagopa/io-app/assets/16268789/7ea38d3a-455e-4787-bfed-0412360a71f9" /> |

## How to test
Run the app with the dev server. Enable the fast login feature and try different scenarios:
- Delay the retrieval of `backendStatus.json` and see that the `IngressScreen` is shown until you get the backend status.
- Try a successful login.
- Try a successful refresh.
- Try to delay server response to produce fetch timeout and after a few attempts you should see the error screen.
